components/validator/l2_output_submitter.go

```diff
@@ -149,49 +146,73 @@ func (l *L2OutputSubmitter) retryAfter(d time.Duration) {
 	})
 }
 
-// TODO(seolaoh): return wait duration explicitly, and handle `retryAfter` function calls at once.
+// trySubmitL2Output l2 Output will continue to check for submittable conditions and try.
+// If it needs to wait, it will check the condition again after the delay.
 func (l *L2OutputSubmitter) trySubmitL2Output(ctx context.Context) error {
-	nextBlockNumber, canSubmit, err := l.CanSubmit(ctx)
+	nextBlockNumber, err := l.FetchNextBlockNumber(ctx)
 	if err != nil {
-		return fmt.Errorf("failed to check if it can submit: %w", err)
+		l.retryAfter(l.cfg.OutputSubmitterRetryInterval)
+		return err
 	}
-	if !canSubmit {
+
+	calculatedWaitTime := l.CalculateWaitTime(ctx, nextBlockNumber)
+	if calculatedWaitTime > 0 {
+		l.retryAfter(calculatedWaitTime)
 		return nil
 	}
 
+	err = l.doSubmitL2Output(ctx, nextBlockNumber)
+	if err != nil {
+		l.retryAfter(l.cfg.OutputSubmitterRetryInterval)
+		return err
+	}
+	l.retryAfter(0)
+	return nil
+}
+
+// doSubmitL2Output Performs the submit operation on the assumption that
+// the l2 Output submit condition is satisfied.
+func (l *L2OutputSubmitter) doSubmitL2Output(ctx context.Context, nextBlockNumber *big.Int) error {
 	output, err := l.FetchOutput(ctx, nextBlockNumber)
 	if err != nil {
-		return fmt.Errorf("failed to fetch next output: %w", err)
+		return err
 	}
 
 	data, err := SubmitL2OutputTxData(l.l2ooABI, output, l.cfg.OutputSubmitterBondAmount)
 	if err != nil {
-		return fmt.Errorf("failed to create submit l2 output transaction data: %w", err)
+		return err
 	}
 
-	if err := l.submitL2OutputTx(data, nextBlockNumber); err != nil {
-		return fmt.Errorf("failed to submit l2 output transaction: %w", err)
+	receipt, err := l.submitL2OutputTx(data)
+	if err != nil {
+		return err
 	}
-	l.metr.RecordL2OutputSubmitted(output.BlockRef)
-	l.retryAfter(l.cfg.OutputSubmitterRetryInterval)
 
+	if receipt.Status != types.ReceiptStatusSuccessful {
+		return fmt.Errorf("failed to send tx with receipt status. TxHash : %+v", receipt.TxHash)
+	}
```

Comment:
This case already returns error in [`txmgr.send()`](https://github.com/kroma-network/kroma/blob/perf/add-buffered-txmgr/utils/service/txmgr/txmgr.go#L260-L263), we don't need to check this case here again!